types/rpi-gpio/index.d.ts

```diff
@@ -0,0 +1,136 @@
+// Type definitions for rpi-gpio 2.1
+// Project: https://github.com/JamesBarwell/rpi-gpio.js#readme
+// Definitions by: Giles Roadnight <https://github.com/Roaders>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// Minimum TypeScript Version: 3.0
+
+/// <reference types="node" />
+import { EventEmitter } from 'events';
+
+type MODE = 'mode_rpi' | 'mode_bcm';
+type DIR = 'in' | 'out' | 'low' | 'high';
+type EDGE = 'none' | 'rising' | 'falling' | 'both';
+type ValueCallback<T> = (err?: Error | null, value?: T) => void;
+type ErrorCallback = (err?: Error | null) => void;
+
+declare class Gpio extends EventEmitter {
```

Comment:
for your consideration, should it be exposed as the class that someone can create? The API seems to be always an already created instance, so here one could expect abstract or interface to describe the contract:
https://github.com/JamesBarwell/rpi-gpio.js/blob/6dd19bf1adf8ba3ac48f13bdc1aa60f4d05341ee/rpi-gpio.js#L507

Reply:
I'm just following what the javascript does. The class is not exported by the javascript so the class won't be available to instantiate.